File: FAST2/Models/LocalMod.vb
Imports System.IO

Namespace Models
    <Serializable()>
    Public Class LocalMod
        Private Sub New()
        End Sub

        Public Sub New(name As String, path As String, Optional author As String = "Unknown", Optional website As String = Nothing)
            Me.Name = name
            Me.Path = path
            Me.Author = author
            Me.Website = website
        End Sub
        
        Public Property Name As String = String.Empty
        Public Property Path As String = String.Empty
        Public Property Author As String = String.Empty
        Public Property Website As String = String.Empty

        Public Shared Function GetLocalMods(Optional serverPathOnly As Boolean = False) As List(Of LocalMod)
            Dim localMods = New List(Of LocalMod)

            Dim foldersToSearch As New List(Of String)

            If serverPathOnly Then
                If Not My.Settings.excludeServerFolder And My.Settings.serverPath IsNot String.Empty
                    foldersToSearch.Add(My.Settings.serverPath)
                End If
            End If
            
            If Not serverPathOnly Then
                For Each folder In My.Settings.localModFolders
                    If folder IsNot Nothing
                        foldersToSearch.Add(folder)
                    End If
                Next
            End If
        
            If foldersToSearch.Count > 0 
              For Each localModFolder In foldersToSearch
                Dim modFolders = Directory.GetDirectories(localModFolder, "@*")

                localMods.AddRange(From modFolder In modFolders Let name = modFolder.Substring(modFolder.LastIndexOf("@", StringComparison.Ordinal) + 1) Let author = "Unknown" Let website = "Unknown" Select New LocalMod(name, modFolder, author, website))
                Next
            End If

            Return localMods
        End Function

    End Class
End NameSpace

File: FASTER/Models/LocalMod.cs
using Microsoft.AppCenter.Crashes;

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

// ReSharper disable UnusedAutoPropertyAccessor.Global

namespace FASTER.Models
{
    [Serializable]
    internal class LocalMod
    {
        private LocalMod()
        { }

        private LocalMod(string name, string path, string author = "Unknown", string website = null)
        {
            Name = name;
            Path = path;
            Author = author;
            Website = website;
        }

        public string Name { get; set; } = string.Empty;
        public string Path { get; set; } = string.Empty;
        public string Author { get; set; } = string.Empty;
        public string Website { get; set; } = string.Empty;

        public static List<LocalMod> GetLocalMods(bool serverPathOnly = false)
        {
            var localMods = new List<LocalMod>();

            List<string> foldersToSearch = new();

            if (serverPathOnly && !string.IsNullOrEmpty(Properties.Settings.Default.serverPath))
            { foldersToSearch.Add(Properties.Settings.Default.serverPath); }

            if (!serverPathOnly && Properties.Settings.Default.localModFolders != null)
            { foldersToSearch.AddRange(Properties.Settings.Default.localModFolders.Where(folder => folder != null && folder != Properties.Settings.Default.serverPath)); }

            if (foldersToSearch.Count <= 0) return localMods;

            foreach (var localModFolder in foldersToSearch)
            {
                try
                {
                    var modFolders = Directory.GetDirectories(localModFolder, "@*");

                    localMods.AddRange(from modFolder in modFolders
                                       let name = modFolder[(modFolder.LastIndexOf("@", StringComparison.Ordinal) + 1)..]
                                       let author = "Unknown"
                                       let website = "Unknown"
                                       select new LocalMod(name, modFolder, author, website));
                }
                catch (Exception e)
                { Crashes.TrackError(e, new Dictionary<string, string> { { "Name", Properties.Settings.Default.steamUserName } }); }
            }

            return localMods;
        }
    }
}
